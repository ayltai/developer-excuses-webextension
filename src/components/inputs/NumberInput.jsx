import { Box, makeStyles, Slider, Typography, } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';

export const NumberInput = props => {
    const classes = makeStyles(theme => ({
        margin : {
            marginBottom : theme.spacing(2),
        },
    }))();

    const [ value, setValue, ] = React.useState(props.value || props.min);

    return (
        <Box className={classes.margin}>
            <Typography variant='body1'>{props.label}</Typography>
            <Slider
                min={props.min}
                max={props.max}
                step={props.step || 1}
                value={value}
                onChange={(event, newValue) => {
                    setValue(newValue);

                    if (props.onChange) props.onChange(newValue);
                }} />
        </Box>
    );
};

NumberInput.propTypes = {
    label    : PropTypes.string,
    min      : PropTypes.number.isRequired,
    max      : PropTypes.number.isRequired,
    step     : PropTypes.number,
    value    : PropTypes.number,
    onChange : PropTypes.func,
};
