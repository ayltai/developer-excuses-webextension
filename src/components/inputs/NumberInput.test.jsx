import { Slider } from '@material-ui/core';
import { mount, shallow, } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import { NumberInput, } from './NumberInput';

describe('<NumberInput />', () => {
    it('renders correctly', () => {
        expect(renderer.create(
            <NumberInput
                label='dummy'
                min={0}
                max={2}
                step={1}
                value={0}
                onChange={() => {}} />
        ).toJSON()).toMatchSnapshot();
    });

    it('mounts without error', () => {
        mount(
            <NumberInput
                label='dummy'
                min={0}
                max={2}
                step={1}
                value={0}
                onChange={() => {}} />
        );
    });

    it('triggers onChange when changed', () => {
        const onChange = jest.fn();

        shallow(
            <NumberInput
                label='dummy'
                min={0}
                max={2}
                step={1}
                value={0}
                onChange={onChange} />)
            .find(Slider)
            .simulate('change', {}, 2);

        expect(onChange.mock.calls.length).toEqual(1);
    });

    it('changes value when changed', () => {
        let value = 0;

        const onChange = newValue => value = newValue;

        shallow(
            <NumberInput
                label='dummy'
                min={0}
                max={2}
                value={value}
                onChange={onChange} />)
            .find(Slider)
            .simulate('change', {}, 2);

        expect(value).toEqual(2);
    });

    it('does not change value when changed', () => {
        let value = 0;

        shallow(
            <NumberInput
                label='dummy'
                min={0}
                max={2}
                step={1}
                value={value} />)
            .find(Slider)
            .simulate('change', {}, 2);

        expect(value).toEqual(0);
    });
});
