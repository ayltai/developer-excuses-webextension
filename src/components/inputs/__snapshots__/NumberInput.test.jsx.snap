// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberInput /> renders correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 makeStyles-margin-1"
>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    dummy
  </p>
  <span
    className="MuiSlider-root MuiSlider-colorPrimary"
    onMouseDown={[Function]}
  >
    <span
      className="MuiSlider-rail"
    />
    <span
      className="MuiSlider-track"
      style={
        Object {
          "left": "0%",
          "width": "0%",
        }
      }
    />
    <input
      type="hidden"
      value="0"
    />
    <span
      aria-orientation="horizontal"
      aria-valuemax={2}
      aria-valuemin={0}
      aria-valuenow={0}
      className="MuiSlider-thumb MuiSlider-thumbColorPrimary"
      data-index={0}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="slider"
      style={
        Object {
          "left": "0%",
        }
      }
      tabIndex={0}
    />
  </span>
</div>
`;
